diff --git a/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntime.cc b/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntime.cc
index 776d25e..1918865 100644
--- a/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntime.cc
+++ b/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntime.cc
@@ -14,7 +14,7 @@
 #define LINE_MAX 2048
 
 static bool memory_region_comparator(MemRange a, MemRange b) {
-  return (a.start < b.start);
+  return (a.start() < b.start());
 }
 
 stl::vector<MemRegion> regions;
@@ -163,7 +163,7 @@ static stl::vector<RuntimeModule> &get_process_map_with_proc_maps() {
       path_buffer[strlen(path_buffer) - 1] = 0;
     }
     strncpy(module.path, path_buffer, sizeof(module.path) - 1);
-    module.load_address = (void *)region_start;
+    module.base = (void *)region_start;
     modules->push_back(module);
 
 #if 0
@@ -191,11 +191,11 @@ static stl::vector<RuntimeModule> get_process_map_with_linker_iterator() {
         if (info->dlpi_name && info->dlpi_name[0] == '/')
           strcpy(module.path, info->dlpi_name);
 
-        module.load_address = (void *)info->dlpi_addr;
+        module.base = (void *)info->dlpi_addr;
         for (size_t i = 0; i < info->dlpi_phnum; ++i) {
           if (info->dlpi_phdr[i].p_type == PT_LOAD) {
             uintptr_t load_bias = (info->dlpi_phdr[i].p_vaddr - info->dlpi_phdr[i].p_offset);
-            module.load_address = (void *)((addr_t)module.load_address + load_bias);
+            module.base = (void *)((addr_t)module.base + load_bias);
             break;
           }
         }
diff --git a/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm b/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
index 7349181..bf0fa74 100644
--- a/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
+++ b/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
@@ -50,8 +50,13 @@ str TMP_REG_0, [sp, #(1 * 8)]
 // call convention: x0 = register context, x1 = interceptor entry
 mov x0, sp
 ldr x1, [sp, #(2 * 8 + 2 * 8 + 30 * 8 + 8 * 16)]
+#if defined(__WIN32__) || defined(__APPLE__)
 adrp TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGE
 add TMP_REG_0, TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGEOFF
+#else
+adrp TMP_REG_0, cdecl(common_closure_bridge_handler)
+add TMP_REG_0, TMP_REG_0, #:lo12:cdecl(common_closure_bridge_handler)
+#endif
 blr TMP_REG_0
 
 // restore stack, saved original sp
